instrumentation/ratpack/ratpack-1.7/library/src/main/java/io/opentelemetry/instrumentation/ratpack/OpenTelemetryHttpClient.java

```diff
@@ -28,7 +28,8 @@ public HttpClient instrument(HttpClient httpClient) throws Exception {
         httpClientSpec -> {
           httpClientSpec.requestIntercept(
               requestSpec -> {
-                Context parentOtelCtx = Context.current();
+                Context parentOtelCtx =
+                    Execution.current().maybeGet(Context.class).orElse(Context.current());
```

Comment:
hi @jsalinaspolo! is this still needed if you are using `RatpackTelemetry.getOpenTelemetryExecInterceptor()`?

Reply:
> is this still needed if you are using `RatpackTelemetry.getOpenTelemetryExecInterceptor()`?

Unforunelty, yes, because the `OpenTelemetryExecInterceptor` only works within Ratpack handlers
The PR helps when we use the Ratpack HttpClient in Ratpack Services, which runs in a different Execution, so the `compute` thread is not blocked.

The Ratpack service will add the OpenTelemetry Context to the `Execution` when Instrumentation is required.

Reply:
is it possible to configure Ratpack Services to use the OpenTelemetryExecInterceptor?

Reply:
> Unforunelty, yes, because the OpenTelemetryExecInterceptor only works within Ratpack handlers

What I meant is that works because the `OpenTelemetryServerHandler` adds the OpenTelemetry Context into the Ratpack Context

> is it possible to configure Ratpack Services to use the OpenTelemetryExecInterceptor?

We could use something similar to `OpenTelemetryServerHandler` but I do not think that it is what we would want 🤔 

The idea of the `OpenTelemetryServerHandler` is to add instrumentation into the `Handlers` (the entry point is an HTTP request) and add the OpenTelemetry Context to the Ratpack Execution, so other instrumentation is propagated within the `Handler`

Ratpack Services, most of the time, you want to run things in the background using something different than the `compute` threads (so the compute threads are not blocked and can receive requests). Hence, many times the Service will be running things in backgrounds like Kafka Consumers, Kafka Streams, Rabbit Consumers, or just background processes with loops. 

Therefore, might not make sense to have a parent span within the Service (as it does with the Handler) but each of the iterations/events running in the service might want to create a `Span` with or without a parent (if the context is propagated in the events, and so on)

I think that for Services, it makes sense to have the flexibility to decide when to create the Span that aggregates the "iteration"